### DISCUSSION TITLE
N+1 query problem how is it solved in Java?

### DISCUSSION THREAD

**Discussion by sbilello** [graphql-java/graphql-java#2257] [General]
Hello!
I was reading this  [article](https://shopify.engineering/solving-the-n-1-problem-for-graphql-through-batching) and I was trying to find the corresponding approach to solve it with GraphQL.

I have a big monolith application based on REST endpoints what's the best approach to convert it to GraphQL?

I was exploring https://graphql-mesh.com/ but I am not sure about it and I would like to understand what kind of problems I might have to solve and if it is able to solve the N+1 query problem behind the scene for me.

Thanks

**Comment by ooga** [comment]
Hello,

the N+1 problem is solved with DataLoader, you can find the official documentation here
https://www.graphql-java.com/documentation/v16/batching/

**Reply by sbilello** [reply]
Thanks for the reply! Is there a full code example for the code in the documentation?
